.github/workflows/ci.yml

```diff
@@ -221,47 +212,51 @@ jobs:
 
     strategy:
       matrix:
-        compiler: ["gcc-13", "clang-18"]
+        os:
+          - "ubuntu-latest"
+        compiler:
+          - "gcc-11"
+          - "gcc-13"
+          - "clang-18"
 
-    runs-on: ubuntu22_4c_16gb
+        include:
+          - os: "windows-2022"
+            compiler: "msvc"
+
+    runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v4
         with:
           lfs: true
 
       - name: General Build Dependencies
+        if: contains(matrix.os, 'ubuntu')
         run: |
-          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
-          sudo apt-get update -q
           sudo apt-get install ninja-build -y -q
 
-      - name: Install GCC
-        if: matrix.compiler == 'gcc-13'
-        run: >-
-          sudo apt-get install ninja-build gcc-13 g++-13 g++-13-multilib -y -q
+      - uses: ./.github/actions/install-gcc
+        if: ${{ contains(matrix.compiler, 'gcc') && contains(matrix.os, 'ubuntu') }}
+        with:
+          version: ${{ matrix.compiler }}
 
-      - name: Install LLVM and Clang
-        if: matrix.compiler == 'clang-18'
-        uses: ./.github/actions/install-clang
+      - uses: ./.github/actions/install-clang
+        if: ${{ contains(matrix.compiler, 'clang') && contains(matrix.os, 'ubuntu') }}
         with:
-          version: "18"
+          version: ${{ matrix.compiler }}
 
       - name: Install vcpkg
         uses: friendlyanon/setup-vcpkg@v1
         with:
           committish: "${{ env.VCPKG_COMMIT }}"
-          cache-version: "test-${{ matrix.compiler }}"
+          cache-version: "test-${{ contains(matrix.os, 'ubuntu') && 'Linux' || 'Windows' }}-${{ matrix.compiler }}"
```

Comment:
i have no idea what string this creates haha

Reply:
'test-Linux-clang-18' or 'test-Windows-mscv' I hope. It's similar to the pattern that the action will create on it's own. But might not be too important, so we could just use our own variables.

I think it is mostly here to be able to cache across the other jobs if they do not have an own cache version (e.g. vcpkg-build has a fallback to test-Linux-clang I think).